### PR TITLE
New check class Y_CHECK_AUNIT_COVERAGE_CHECK

### PR DESCRIPTION
**Pull request by ThomasKempfle**
I've added a new check class for ABAP unit test coverage
(refere to issue https://github.com/SAP/code-pal-for-abap/issues/83)
I use it here at my employer to make sure, that other developers produce a minium level of unit tests with their working code.

Since this is my first pull request and I have not contributed yet, please be lenient with every deviation from existing guidelines/architectural patterns. 

What is missing/needs to be done: 
> The docu page ( I've created the docu as a popup in my original class and replaced it with the Code Pal pattern)
> The complexity logic is not perfect yet: It should prevent messages for sources with a low complexity. However: 
The test coverage is currently measured on the average of all method in a class and the complexity is currently measured by source code object. So if you have one class with 100 GETer/SETer methods and only one complex method, you might have to produce unit tests for the SETer/GETers also, in order to pass the test.

### CODE REVIEW

File: clean_code_main/clean_code_checks/y_check_base.clas.xml [hunk 1]
```diff
@@ -12,56 +12,6 @@
...
-     <ENTRY>Choose the Type of Code to be checked</ENTRY>
-     <LENGTH>60</LENGTH>
-    </item>
-    <item>
-     <ID>I</ID>
-     <KEY>301</KEY>
-     <ENTRY>Choose a Message Severity</ENTRY>
-     <LENGTH>50</LENGTH>
-    </item>
-   </TPOOL>
```
Comment (lucasborin) [inline clean_code_main/clean_code_checks/y_check_base.clas.xml:64]: why did you remove the text element?

Comment (lucasborin) [reply clean_code_main/clean_code_checks/y_check_base.clas.xml:64]: it looks like you removed the text elements from 35 classes

Comment (eugen-guenther-sap) [reply clean_code_main/clean_code_checks/y_check_base.clas.xml:64]: if you change 38 Files to release one check we cannot do that.
the base class was changed, too. (please don't do that if you just want to release one check)

Comment (ThomasKempfle) [reply clean_code_main/clean_code_checks/y_check_base.clas.xml:64]: ??? I did not touched other classes than my newly created one !!! My workflow have been:
- Fork the original repository into a new repository
- Download the DEV branch to one of my development systems using the ABAP GIT client
- Add the new class to the same package
- Stage the package
- Create a Pull request
In that process, I did not changed any of the other classes. So this seems to be an issue of the GITclient and my development system....

Comment (eugen-guenther-sap) [reply clean_code_main/clean_code_checks/y_check_base.clas.xml]: ok, now it looks better than before.
